app/models/miq_report_result.rb

```diff
@@ -35,6 +35,9 @@ class MiqReportResult < ApplicationRecord
     end
   end
 
+  before_save  { @_extra_groupings = report.extras.delete(:grouping) }
+  after_commit { report.extras[:grouping] = @_extra_groupings }
```

Comment:
Worth noting:  This has to be an `after_commit` instead of a `after_save`, otherwise it puts the `ActiveRecord` object into a dirty state, and causes an infinite loop.  If this gets changed to the latter, the test should catch it.

Reply:
What is this witchcraft... So, are we removing it before we commit it to the DB but put it back into the in-memory object?

Reply:
Should we clear `@_extra_groupings` after?

Reply:
> So, are we removing it before we commit it to the DB but put it back into the in-memory object?

Yup, basically.  Trying to play it safe just incase I missed something where this is used after it is saved.

> Should we clear @_extra_groupings after?

I had that in there previously, but nothing is using that var, so I am tempted to just not bother.  We aren't `.dup`ing it, so just passing a reference of it around.  Up to you though if you feel strongly one way or the other.